src/avram/delete_operation_template.cr

```diff
@@ -0,0 +1,21 @@
+class Avram::DeleteOperationTemplate
+  macro setup(type, columns, *args, **named_args)
+
+    # This makes it easy for plugins and extensions to use the base SaveOperation
+    def base_query_class : ::{{ type }}::BaseQuery.class
+      ::{{ type }}::BaseQuery
+    end
+
+    def delete_operation_class : ::{{ type }}::DeleteOperation.class
+      ::{{ type }}::DeleteOperation
+    end
+
+    class ::{{ type }}::DeleteOperation < Avram::DeleteOperation({{ type }})
+      macro inherited
+        FOREIGN_KEY = "{{ type.stringify.underscore.id }}_id"
```

Comment:
I don't think I'm using this. My thought was maybe we need to know the foreign key to delete an associated record?  But the database probably handles it directly 🤔